src/Models/ShallowWaterModels/rk3_substep_shallow_water_model.jl

```diff
@@ -57,8 +57,8 @@ Time step solution fields with a 3rd-order Runge-Kutta method.
     i, j, k = @index(Global, NTuple)
 
     @inbounds begin
-        U.uh[i, j, k] += Δt * γ¹ * G¹.uh[i, j, k]
-        U.vh[i, j, k] += Δt * γ¹ * G¹.vh[i, j, k]
-        U.h[i, j, k]  += Δt * γ¹ * G¹.h[i, j, k]
+        U[1][i, j, k] += Δt * γ¹ * G¹[1][i, j, k]
```

Comment:
Loop?

Reply:
I like it written out.

The main thing to fix here are the errors in units in advection terms etc